drivers/sensor/shtcx/shtcx.h

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2021 Thomas Stranger
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_SENSOR_SHTCX_SHTCX_H_
+#define ZEPHYR_DRIVERS_SENSOR_SHTCX_SHTCX_H_
+
+#include <device.h>
+#include <devicetree.h>
+#include <kernel.h>
+#include <drivers/gpio.h>
+
+/* common cmds */
+#define SHTCX_CMD_READ_ID              0xEFC8
+#define SHTCX_CMD_SOFT_RESET           0x805D
+/* shtc3 only: */
+#define SHTCX_CMD_SLEEP                0xB098
+#define SHTCX_CMD_WAKEUP               0x3517
+
+#define SHTCX_POWER_UP_TIME_US   240U
+/* Soft reset time is 230us for shtc1 and 240us for shtc3 */
+#define SHTCX_SOFT_RESET_TIME_US 240U
+
+#define SHTCX_MAX_READ_LEN 6
+#define SHTCX_WORD_LEN 2
+#define SHTCX_CRC8_LEN 1
+
+#define SHTC3_ID_MASK    0x083F
+#define SHTC3_ID_VALUE   0x0807
+#define SHTC1_ID_MASK    0x083F
+#define SHTC1_ID_VALUE   0x0007
+
+#define CHIP_SHTC1 0
+#define CHIP_SHTC3 1
+#define TO_CHIP(inst) \
+	CONCAT(CHIP_, DT_ENUM_UPPER_TOKEN(DT_DRV_INST(inst), chip))
+
+#define MEASURE_MODE_NORMAL 0
+#define MEASURE_MODE_LOW_POWER 1
+#define TO_MEASURE_MODE(inst)  CONCAT(MEASURE_MODE_, \
+			   DT_ENUM_UPPER_TOKEN(DT_DRV_INST(inst), measure_mode))
+
+enum shtcx_chip {
+	SHTC1 = CHIP_SHTC1,
+	SHTC3 = CHIP_SHTC3,
+};
+
+enum shtcx_measure_mode {
+	NORMAL = MEASURE_MODE_NORMAL,
+	LOW_POWER = MEASURE_MODE_LOW_POWER,
+};
+
+struct shtcx_sample {
+	uint16_t temp;
+	uint16_t humidity;
+} __packet;
```

Comment:
Does the `build_all` change really build this? It looks like this should cause a build error because it should be `__packed`, not `__packet`.

Reply:
Thanks! Good find. Using west and the zephyr-sdk I didn't get a warning, and as the members were correctly aligned on my test platform I didn't notice it.
After correcting it to __packed, the struct was aligned to an odd address and curiously the mcu got stuck at the first write to it. Therefore I have additionally added __aligned(2). So this should work on all platforms now.